tests/spellchecking/spellcheck.js

```diff
@@ -0,0 +1,275 @@
+/* bender-tags: editor,unit */
+/* bender-ckeditor-plugins: nanospell */
+
+'use strict';
+
+(function () {
+	bender.editor = {
+		config: {
+			enterMode: CKEDITOR.ENTER_P
+		}
+	};
+
+  bender.test({
+    assertHtml: function( expected, actual, msg ) {
+			assert.areEqual( bender.tools.fixHtml( expected ), bender.tools.fixHtml( actual ), msg );
+		},
+		setUp: function () {
+			this.server = sinon.fakeServer.create();
+			this.server.respondImmediately = true;
+
+			// for these tests we don't really care that much about the
+			// mock data, just that it returns something vaguely resembling the server call
+			var suggestions = {
+				"result": {
+					"appkes": ["apples"],
+					"pearrs": ["pears"],
+					"bannanas": ["bananas"],
+          "missspelling": ["misspelling"]
+				}
+			};
+
+			this.server.respondWith(
+				'/spellcheck/nano/',
+				JSON.stringify(suggestions)
+			);
+		},
+		tearDown: function () {
+			this.server.restore();
+			// reset the plugin and clear all spellcaches
+			this.editorBot.editor.execCommand('nanospellReset');
+		},
+    'test it can spellcheck a simple paragraph': function() {
+			var bot = this.editorBot,
+				tc = this,
+				editor = bot.editor,
+				resumeAfter = bender.tools.resumeAfter,
+				starterHtml = '<p>Paragraph with missspelling</p>';
+
+			bot.setHtmlWithSelection(starterHtml);
+
+			resumeAfter(editor, 'spellCheckComplete', function() {
+				var paragraph = editor.editable().findOne('p');
+
+				tc.assertHtml('<p>Paragraph with <span class="nanospell-typo">missspelling</span></p>', paragraph.getOuterHtml());
+			});
+
+			wait();
+		},
+    'test it can spellcheck across inline style elements': function() {
+      var bot = this.editorBot,
+				tc = this,
+				editor = bot.editor,
+				resumeAfter = bender.tools.resumeAfter,
+				starterHtml = '<p>Paragraph with mis<b>s</b>s<i>pe</i>lling</p>';
+
+      bot.setHtmlWithSelection(starterHtml);
+
+			resumeAfter(editor, 'spellCheckComplete', function() {
+				var paragraph = editor.editable().findOne('p');
+
+				tc.assertHtml('<p>Paragraph with <span class="nanospell-typo">mis<b>s</b>s<i>pe</i>lling</span></p>', paragraph.getOuterHtml());
+			});
+
+			wait();
+    },
+    'test it can spellcheck across line breaks': function() {
+      var bot = this.editorBot,
+				tc = this,
+				editor = bot.editor,
+				resumeAfter = bender.tools.resumeAfter;
+
+      bot.setHtmlWithSelection(
+        '<p>appkes<br>' +
+          'pearrs<br>' +
+          'bannanas' +
+        '</p>'
+      );
+
+      resumeAfter(editor, 'spellCheckComplete', function() {
+        var paragraph = editor.editable().findOne('p');
+
+        tc.assertHtml(
+          '<p><span class="nanospell-typo">appkes</span><br>' +
+            '<span class="nanospell-typo">pearrs</span><br>' +
+            '<span class="nanospell-typo">bannanas</span>' +
+          '</p>',
```

Comment:
going back to my previous comment, maybe we do care about the html here (breaks are not actually wrapped in spans by accident, etc)